import_logs.py

```diff
@@ -1474,16 +1442,15 @@ def _call(path, args, headers=None, url=None, data=None):
             https_handler_args = {'context': ssl_context}
         else:
             https_handler_args = {}
-        opener = urllib2.build_opener(
-            Matomo.RedirectHandlerWithLogging(),
-            urllib2.HTTPSHandler(**https_handler_args))
+        opener = urllib.request.build_opener(
+            self.RedirectHandlerWithLogging(),
+            urllib.request.HTTPSHandler(**https_handler_args))
         response = opener.open(request, timeout = timeout)
         result = response.read()
```

Comment:
You need to .decode() the response, as otherwise it would be returned as bytes, which e.g. json.loads() cannot use.

Reply:
@cbay  are you able to point out a test case where the current implementation would fail? Tests are currently running, so would be good to add a test if possible...

Reply:
The current test suite only tests a small part of the import script. It doesn't actually try to import hits to a Matomo instance, which is where the issue lies.

Actually adding such tests would be a massive task, I'm afraid.

Reply:
Oh. I see. Would that be covered by the tests done on Matomo side? It uses the log importer in some test fixtures to import log files: https://github.com/matomo-org/matomo/blob/3.x-dev/tests/PHPUnit/Fixtures/ManySitesImportedLogs.php#L110

Reply:
Yes, I think so!

I've also tested the script myself today (with the 2 aforementioned changes), it seems to work fine, at least for my use case. It has successfully parsed/recorded several millions lines. Using Python 3.4.

Reply:
awesome. Would you mind to add your changes as suggestions, so we can simply apply them?

Reply:
I didn't know you could do suggestions on code reviews! That's nice.